openupgrade_scripts/scripts/sale/16.0.1.2/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,97 @@
+---Models in module 'sale'---
+obsolete model report.all.channels.sales [sql_view]
+# DONE: pre-migration: drop view
+
+obsolete model report.sale.report_saleproforma [abstract]
+# NOTHING TO DO: obsolete model, does not exist in the database
+
+obsolete model sale.payment.acquirer.onboarding.wizard [transient]
+new model sale.payment.provider.onboarding.wizard [transient]
+# DONE: pre-migration: renamed model sale.payment.acquirer.onboarding.wizard to sale.payment.provider.onboarding.wizard
+
+---Fields in module 'sale'---
+sale         / account.analytic.applicability / business_domain (False)       : NEW selection_keys: ['bill', 'expense', 'general', 'invoice', 'purchase_order', 'sale_order'], mode: modify
+# NOTHING TO DO
+
+sale         / account.bank.statement.line / partner_shipping_id (many2one): module is now 'account' ('sale')
+sale         / account.move             / partner_shipping_id (many2one): module is now 'account' ('sale')
+sale         / account.move.line        / is_downpayment (boolean)      : NEW
+sale         / account.payment          / partner_shipping_id (many2one): module is now 'account' ('sale')
+# NOTHING TO DO
+
+sale         / payment.acquirer         / so_reference_type (selection) : DEL selection_keys: ['partner', 'so_name']
+sale         / payment.provider         / so_reference_type (selection) : NEW selection_keys: ['partner', 'so_name'], hasdefault: default
+# NOTHING TO DO
+
+sale         / sale.order               / show_update_fpos (boolean)    : NEW
+sale         / sale.order               / show_update_pricelist (boolean): not stored anymore
+# NOTHING TO DO: handle by ORM
+
+sale         / sale.order.line          / analytic_distribution_stored_char (char): NEW isfunction: function, stored
+# NOTHING TO DO: handle by ORM
+
+sale         / sale.order.line          / analytic_tag_ids (many2many)  : DEL relation: account.analytic.tag
+# DONE: fast fill analytic_distribution in pre-migration
+
+sale         / sale.order.line          / product_type (selection)      : previously in module sale_stock
+# NOTHING TO DO: handle by ORM
+
+sale         / sale.order.line          / qty_delivered (float)         : not a function anymore
+# NOTHING TO DO: only the inverse compute function was removed
+
+sale         / sale.order.line          / qty_delivered_manual (float)  : DEL
+# NOTHING TO DO: not used any more, this field is useless since the new compute method from 13.0
+
+---XML records in module 'sale'---
+NEW ir.actions.act_window: sale.action_open_sale_payment_provider_onboarding_wizard
+DEL ir.actions.act_window: sale.action_open_sale_onboarding_payment_acquirer_wizard
+DEL ir.actions.act_window: sale.action_sale_order_form_view
+# NOTHING TO DO: noupdate="0" records
+
+ir.actions.server: sale.model_sale_order_action_share (noupdate switched)
+# DONE: pre-migration: noupdate switched to 0
+
+NEW ir.config_parameter: sale.default_invoice_email_template (noupdate)
```

Comment:
Hi @hoangtiendung070797 thanks for your work on this PR !
Actually this one is not new, it was already existing in v15.0 here : https://github.com/odoo/odoo/blob/7d90bce4c90cb77f2268332a5f1225cd884ac674/addons/sale/models/res_config_settings.py#L59-L65 (except it was not loaded from data but directly in Python model).

Obviously when migrating I get the following errors : 
```
2023-10-12 12:15:55,539 65829 INFO lf_20231011_v16mig odoo.modules.loading: loading sale/data/sale_data.xml 
2023-10-12 12:15:55,556 65829 ERROR lf_20231011_v16mig odoo.sql_db: bad query: INSERT INTO "ir_config_parameter" ("create_date", "create_uid", "key", "value", "write_date", "write_uid") VALUES ('2023-10-12 12:15:52.952317', 1, 'sale.default_invoice_email_template', '11', '2023-10-12 12:15:52.952317', 1) RETURNING "id"
ERROR: duplicate key value violates unique constraint "ir_config_parameter_key_uniq"
DETAIL:  Key (key)=(sale.default_invoice_email_template) already exists.
```